okhttp/src/main/kotlin/okhttp3/internal/tls/TrustManagerJvm.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.tls
+
+import android.annotation.SuppressLint
+import java.net.Socket
+import java.security.cert.X509Certificate
+import javax.net.ssl.SSLEngine
+import javax.net.ssl.X509ExtendedTrustManager
+import javax.net.ssl.X509TrustManager
+import org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
+
+@IgnoreJRERequirement
+@SuppressLint("NewApi")
+class TrustManagerJvm internal constructor(
```

Comment:
why do these need to be separate types? Can’t we implement once for all? Does the Android one break on the JVM?

Reply:
Different class hierarchies, different methods, different semantics.  On JVM we need it to implement X509ExtendedTrustManager, on Android we can't even rely on that type, and those methods can never be called.